File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # # Solution 1
        # if len(s) != len(t):
        #     return False
        # my_dict = {}
        # for char in s:
        #     if char in my_dict:
        #         my_dict[char] +=1
        #     else:
        #         my_dict[char] = 1
        # for char in t:
        #     if char in my_dict and my_dict[char] != 0:
        #         my_dict[char] -= 1
        #     else:
        #         return False
        # return True

        # Solution 2
        return Counter(s) == Counter(t)

File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> dict = new();
        if (s.Length != t.Length) return false;
        foreach(char c in s)
            if (dict.ContainsKey(c))
                dict[c] ++;
            else
                dict[c] = 1;
        foreach(char c in t)
            if (dict.ContainsKey(c) && dict[c] > 0)
                dict[c] --;
            else
                return false;
        return true;
    }
}